lib/src/component_declaration/component_base/component_base_2.dart

```diff
@@ -72,25 +72,55 @@ abstract class UiComponent2<TProps extends UiProps> extends react.Component2
   /// > Should be used alongside [forwardingClassNameBuilder].
   ///
   /// > Related [copyUnconsumedDomProps]
+  ///
+  /// __Deprecated.__ Use [addUnconsumedProps] within
+  /// [modifyProps] (rather than [addProps]) instead. Will be removed in the
+  /// `4.0.0` release.
   @override
+  @Deprecated('4.0.0')
   Map copyUnconsumedProps() {
     var consumedPropKeys = consumedProps?.map((ConsumedProps consumedProps) => consumedProps.keys) ?? const [];
 
     return copyProps(keySetsToOmit: consumedPropKeys);
   }
 
+  /// Passes a reference of a component's [UiProps] to be updated with unconsumed props.
+  ///
+  /// > Related [addUnconsumedDomProps]
+  void addUnconsumedProps(Map props) {
+    var consumedPropKeys = consumedProps?.map((ConsumedProps consumedProps) => consumedProps.keys);
+
+    forwardUnconsumedProps(this.props, propsToUpdate: props,
+        keySetsToOmit: consumedPropKeys);
+  }
+
   /// Returns a copy of this component's props with keys found in [consumedProps] and non-DOM props omitted.
   ///
   /// > Should be used alongside [forwardingClassNameBuilder].
   ///
   /// > Related [copyUnconsumedProps]
+  ///
+  /// __Deprecated.__ Use [addUnconsumedDomProps] within
+  /// [modifyProps] (rather than [addProps]) instead. Will be removed in the
+  /// `4.0.0` release.
   @override
+  @Deprecated('4.0.0')
   Map copyUnconsumedDomProps() {
     var consumedPropKeys = consumedProps?.map((ConsumedProps consumedProps) => consumedProps.keys) ?? const [];
 
     return copyProps(onlyCopyDomProps: true, keySetsToOmit: consumedPropKeys);
   }
 
+  /// Passes a reference of a component's [UiProps] to be updated with the unconsumed DOM props.
```

Comment:
Good call! That's much more clear. Wouldn't we want the `modifyProps` line to actually be `..modifyProps(addUnconsumedDomProps)` though, @aaronlademann-wf? `addUnconsumedDomProps` is void because `modifyProps` uses it as a callback and passes in  itself for `props` parameter

Reply:
derp derp, yes.